### DISCUSSION TITLE
Lua help needed with MPV WLX plugin

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1827] [General]
Linux Debian based distro
`MPV_alt` WLX plugin

I am trying to be able to make simple modifications to the MPV OSC used in the WLX plugin called `MPV_alt`
I am aware that this has to do with MPV but it is also a WLX plugin for DC.

This osc.lua file can be found in the source code for MPV found here
https://github.com/mpv-player/mpv/blob/master/player/lua/osc.lua

I have been able to figure out more than I thought I would, but there is just one issue I cannot seem to figure out.
 
I would like my simple custom OSC to look like the following
`00:00:00 |------------------------------SEEKBAR------------------------------| 00:00:00`
I have been able to achieve this already by crudely removing all the icons (did this in a weird way but appears to have worked). However the timecodes + seekbar mentioned above are not centered by default in the default OSC.  Just looking for some help on how to center these items.

`OSC SEEKBAR & TIMECODES - CENTERING`
OPTION 1
The minimal thing I would like is to be able to just move the SEEKBAR + TIMECODES altogether and CENTER them in the OSC.

OPTION 2
With this option I would like to be able to do the same as OPTION 1 but I would like to be able to....
-Move each TIMECODE as far LEFT & RIGHT sides as I could
-Then be able to change the width of the SEEKBAR in between them to fill in the space
00:00:00 |------------------------------SEEKBAR------------------------------| 00:00:00

Thank you to anyone willing to look into this and for any help provided.
Hopefully someone can help because this does not appear to be common knowledge.

**Comment by Skif-off** [comment]
You chose a very strange place to ask the question :)) Why not, for example, [here](https://github.com/mpv-player/mpv/discussions)?

> I figured I would ask this here because many DC users possibly use Lua which these next questions require.

It's good that you don't ask questions about DC to mpv users, because they also use Lua ;) (By default mpv is compiled with Lua support, OSC is enabled by default, so **almost all** mpv users use Lua :))

> I am pretty sure that there are only 4-5 people on the planet who can understand how to modify this Lua file for MPV :)

Maybe :))) Also some people write third-party OSC, but communicate... on mpv discussion platforms.

> I am trying to be able to edit the OSC used in the WLX plugin called `MPV_alt`

Maybe it would be better to make changes in the regular player and then transfer them to the plugin?

As I understand, some parameters of `osc.lua` can be changed with `osc.conf`. Maybe these parameters can be changed with `plugload.lua`? (I have seen examples of accessing `script-opts`, but I don't know if they can be changed on the fly.)
